packages/pwa-kit-dev/src/ssr/server/build-dev-server.js

```diff
@@ -126,6 +127,44 @@ export const DevServerMixin = {
         })
     },
 
+    /**
+     * @private
+     */
+    _setupExtensions(app, options) {
+        // TODO: support extensions options array syntax i.e. ['extension-a', {}]
+        const extensions = options.mobify?.app?.extensions || []
+        app.__extensions = extensions || []
+
+        extensions.forEach((extension) => {
+            const setupServerFilePathBase = path.join(
+                options.projectDir,
+                'node_modules',
+                extension,
+                'setup-server'
+            )
+
+            let filePath
+            if (fs.existsSync(`${setupServerFilePathBase}.ts`)) {
+                filePath = `${setupServerFilePathBase}.ts`
+            } else if (fs.existsSync(`${setupServerFilePathBase}.js`)) {
+                filePath = `${setupServerFilePathBase}.js`
+            } else {
+                return
+            }
+
+            const setupServer = tsx.require(filePath, __filename)
```

Comment:
You could probably clean this up a little by doing something like this:

```
let setupServer
const foundType = SUPPORTED_FILE_TYPES.find((type) => fs.existsSync(`${setupServerFilePathBase}.${type}`))

if (foundType) {
    tsx.require(`${setupServerFilePathBase}.${foundType}`, __filename)
} else {
    return
}
```

The advantage here is that we have a constant to of supported files that makes it easier to support more in the future (if that is even a think) but also makes it easier to change or make it readable that you have precedence of TS over TS

Reply:
beautiful, thank you